packages/astro/src/vite-plugin-astro/index.ts

```diff
@@ -144,8 +141,13 @@ export default function astro({ config, logging }: AstroPluginOptions): vite.Plu
 				});
 
 				// Signal to Vite that we accept HMR updates
-				const SUFFIX = isProduction ? '' : `\nif (import.meta.hot) import.meta.hot.accept((mod) => mod);`;
-
+				let SUFFIX = '';
+				if (isProduction) {
+					SUFFIX += `\nif (import.meta.hot) import.meta.hot.accept((mod) => mod);`;
+				}
+				if (isPage) {
+					SUFFIX += `\nimport "$scripts/page.js";`;
+				}
```

Comment:
What is this doing? I'd prefer to keep this plugin agnostic of pages vs components if we can.